Rpackage/R/mmif.R

```diff
@@ -30,58 +29,46 @@
 #' colnames(feature.mat) <- c("a", "b", "c")
 #' feature.mat <- data.frame(feature.mat)
 #' 
-#' trees <- mmif(target.mat, feature.mat, margin = 2.0, n_cpu = -1)
+#' trees <- mmif(target.mat, feature.mat, margin = 2.0)
 #' 
-#' @export
 mmif <- structure(function(target.mat, feature.mat, 
                            max_depth = Inf, margin=0.0, loss="hinge",
                            min_sample = 1, n_trees = 10,
-                           n_features =  ceiling(ncol(feature.mat)**0.5), n_cpu = 1){
+                           n_features =  ceiling(ncol(feature.mat)**0.5)){
   
-  
-  ### parallelize using foreach, see all permutation combination of param grid values
-  ### register parallel backend
-  if(n_cpu == -1) n_cpu <- detectCores() 
-  assert_that(detectCores() >= n_cpu)
+  #lapply parallel or sequencial
+  Lapply <- if(requireNamespace("future.apply")){ 
+    future.apply::future_lapply 
+  }
+  else{ lapply }
   
   all_trees <- list()
-  if(n_cpu > 1){
-    cl <- makeCluster(n_cpu)
-    registerDoParallel(cl)
     
-    ### create n_trees
-    all_trees <- foreach(i = 1 : n_trees, .packages = "mmit") %dopar% 
-      random_tree(target.mat, feature.mat, 
+  ### create n_trees
+  all_trees <- Lapply(1 : n_trees, function(x) .random_tree(target.mat, feature.mat, 
                   max_depth = max_depth, margin = margin, loss = loss,
                   min_sample = min_sample, n_trees = n_trees,
-                  n_features = n_features)
-    
-    
-    stopCluster(cl)
-  }
-  else{
-    for(i in 1 : n_trees) {
-      all_trees[[i]] = random_tree(target.mat, feature.mat, 
-                                   max_depth = max_depth, margin = margin, loss = loss,
-                                   min_sample = min_sample, n_trees = n_trees,
-                                   n_features = n_features)
-    }
+                  n_features = n_features))
     
-  }
   
   return(all_trees)
   
 }, ex=function(){
   
-  data(neuroblastomaProcessed, package="penaltyLearning")
+  data("neuroblastomaProcessed", package="penaltyLearning", envir=environment())
   feature.mat <- data.frame(neuroblastomaProcessed$feature.mat)[1:45,]
   target.mat <- neuroblastomaProcessed$target.mat[1:45,]
+  if(require(parallel)){
+    cl <- makeCluster(detectCores())
+    registerDoParallel(cl)
```

Comment:
Shouldn't this create a WARNING or ERROR because registerDoParallel is in the foreach package, which is no longer imported?

Maybe the Rd files need to be re-generated via `inlinedocs::package.skeleton.dx()` ?

anyways please delete this if statement since foreach is no longer used

Reply:
ok